sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3510,6 +3526,20 @@ def _teardown(self,
             else:
                 raise
 
+        if handle.launched_resources.cloud.is_same_cloud(clouds.Azure()):
+            task_use_az_ml = skypilot_config.get_nested(('azure', 'use_az_ml'),
+                                                        False)
+            cluster_use_az_ml = handle.launched_resources.use_az_ml
+            if cluster_use_az_ml != task_use_az_ml:
+                task_str = 'uses' if task_use_az_ml else 'does not use'
+                cluster_str = 'uses' if cluster_use_az_ml else 'does not use'
+                with ux_utils.print_exception_no_traceback():
+                    raise exceptions.ResourcesMismatchError(
+                        f'Current setup {task_str} Azure ML API, but the '
+                        f'specified cluster {cluster_name} to terminate '
+                        f'{cluster_str} it. Please set azure.use_az_ml '
+                        f'to {cluster_use_az_ml} in ~/.sky/config.yaml.')
+
```

Comment:
Same as above, we should just allow the teardown using the non-ML API if the existing cluster was launched using non-ML API